spec/system/admin/enterprises_spec.rb

```diff
@@ -725,6 +725,53 @@
             .to eq('Enterprise "First Distributor" has been successfully updated!')
           expect(distributor1.reload.hide_groups_tab).to be false
         end
+
+        context "creating custom tabs" do
+          before do
+            visit edit_admin_enterprise_path(distributor1)
+            within(".side_menu") do
+              click_link "White Label"
+            end
+            check "Create custom tab in shopfront"
+          end
+
+          it "display a form with the custom tab fields: title and content" do
+            expect(page).to have_content "TITLE FOR CUSTOM TAB"
+            expect(page).to have_field "enterprise_custom_tab_attributes_title"
+            expect(page).to have_content "CONTENT FOR CUSTOM TAB"
+            expect(page).to have_field "enterprise_custom_tab_attributes_content"
+          end
+
+          it "can save custom tab fields" do
+            fill_in "enterprise_custom_tab_attributes_title", with: "Custom tab title"
+            fill_in "enterprise_custom_tab_attributes_content", with: "Custom tab content"
+            click_button 'Update'
+            expect(flash_message)
+              .to eq('Enterprise "First Distributor" has been successfully updated!')
+            expect(distributor1.reload.custom_tab.title).to eq("Custom tab title")
+            expect(distributor1.reload.custom_tab.content).to eq("Custom tab content")
+          end
+
+          context "when custom tab is already created" do
+            let(:custom_tab) { create(:custom_tab) }
+
+            before do
+              distributor1.update(custom_tab: custom_tab)
+              visit edit_admin_enterprise_path(distributor1)
+              within(".side_menu") do
+                click_link "White Label"
+              end
+            end
+
+            it "display the custom tab fields with the current values" do
+              expect(page).to have_checked_field "Create custom tab in shopfront"
+              expect(page).
+                to have_field "enterprise_custom_tab_attributes_title", with: custom_tab.title
+              expect(page).
+                to have_field "enterprise_custom_tab_attributes_content", with: custom_tab.content
+            end
```

Comment:
What are the current values? If they are empty then this test may not work and we don't know. It's better to define test data and then test against that text explicitely.

Reply:
Right: a296ba2ec3 !